### PR TITLE
add validation for dimensions

### LINKED ISSUE #645: gray out dimension field (and remove the dropdown / replace with dimension text) if metrics def no longer has that dimension
If I have a metrics config, and I go back to the model and drop a column that was used as a dimension, the dropdown is completely empty. I think I would expect the text to gray out and not be able to select that field.

Screen Shot 2022-07-25 at 3 23 55 PM

### PR DESCRIPTION
**Pull request by bcolloran**
fixes #645

### CODE REVIEW

File: src/lib/components/table-editable/TableCellSelector.svelte [hunk 1]
```diff
@@ -33,13 +47,23 @@
...
   };
 
-  $: options = columnConfig.cellRenderer.options;
+  let tdElt: HTMLTableCellElement;
+  let tdHeight: number = undefined;
+  onMount(() => {
+    tdHeight = tdElt.getBoundingClientRect().height;
+  });
+  const ALERT_CIRCLE_SIZE_PX = 20;
+  $: alertCircleTop = tdHeight / 2 - ALERT_CIRCLE_SIZE_PX / 2;
```
Comment (bcolloran) [inline src/lib/components/table-editable/TableCellSelector.svelte:56]: i know it's a little weird to set the positioning of this with JS by checking on the height of the container, but this approach fits with some other plans I have for the table layout. (And I guess in any case it's probably temporary while we work on our component sets)

Comment (hamilton) [reply src/lib/components/table-editable/TableCellSelector.svelte:56]: what's the intention here?

**Comment by bcolloran** [comment]
@hamilton tagging you for review on this one re #645

**Comment by magorlick** [comment]
Thank you @bcolloran , the feedback is much improved on the config surface.  I did notice, however, that the dashboard is still presenting the missing dimension with "no available values". I think we should probably remove leaderboards and dimensions that are not valid. It is not clear if all the values are null or the dimension is not available at the dashboard level.

**Review by AdityaHegde — APPROVED** [review]

**Comment by bcolloran** [comment]
@magorlick -- i think that goes beyond the scope of #645, can we merge this as an incremental improvement and file those additional observations as a new issue? If so, would you mind clicking the "approve" button on this one so that we can get it in? Thanks :-)

**Review by magorlick — APPROVED** [review]

**Pull request merged**